src/ngx_http_lua_tcpcong.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) Rain Li (blacktear23)
+ *
+ */
+
+#ifndef DDEBUG
+#define DDEBUG 0
+#endif
+#include "ddebug.h"
+
+
+#include "ngx_http_lua_tcpcong.h"
+#include "ngx_http_lua_util.h"
+#include "ngx_http_lua_api.h"
+
+
+int ngx_socket_set_tcp_congestion(ngx_socket_t s, const char* cong_name);
+static int ngx_http_lua_ngx_set_tcp_congestion(lua_State *L);
+
+
+#if (NGX_LINUX)
+
+int
+ngx_socket_set_tcp_congestion(ngx_socket_t s, const char* cong_name) {
+    u_char optval[16];
+    ngx_cpystrn(optval, (u_char *)cong_name, 16);
```

Comment:
We need a space after the type casting expression `(u_char *)`.

Also, i don't think we need to copy a new buffer here in the first place.